lib/tests/unit/modules/task_test.cc

```diff
@@ -207,6 +207,32 @@ TEST_CASE("Modules::Task::executeAction", "[modules][output]") {
             0 };
         ActionRequest request { RequestType::Blocking, echo_content };
 
+        auto output = e_m.executeAction(request).action_metadata.get<std::string>({"results", "stdout"});
+        boost::trim(output);
+        std::cout << output;
+        REQUIRE(output == "{\"message\":\"hello\",\"_task\":\"test\"}");
+    }
+
+    SECTION("builds installdir") {
+        auto echo_txt =
+#ifndef _WIN32
+            (DATA_FORMAT % "\"0632\""
+                         % "\"task\""
+                         % "\"run\""
+                         % "{\"task\": \"test\", \"input_method\": \"environment\",  \"input\":{\"message\":\"hello\"}, \"metadata\": {\"files\": [\"file1.txt\", \"dir/\"]}, \"files\" : [{\"sha256\": \"c9b0792365dddb71eca56e722aa8913a8de1721c2b3ee5fdbd697a9da988ab3f\", \"filename\": \"init.sh\"}, {\"sha256\": \"67ee5478eaadb034ba59944eb977797b49ca6aa8d3574587f36ebcbeeb65f70e\", \"filename\": \"dir/file2.txt\"}, {\"sha256\": \"ecdc5536f73bdae8816f0ea40726ef5e9b810d914493075903bb90623d97b1d8\", \"filename\": \"file1.txt\"}]}").str();
```

Comment:
Will clean up formatting and account for windows. 
I am adding the required resources in the src/test/resources/task-cache directory saved under their hashes. Will update once i get the trailing directory support added.